raiden/network/pathfinding.py

```diff
@@ -1,19 +1,107 @@
+import random
+import sys
+from typing import Optional
+
+import click
 import requests
+import structlog
 from eth_utils import to_hex
 
-from raiden.constants import DEFAULT_HTTP_REQUEST_TIMEOUT
+from raiden.constants import DEFAULT_HTTP_REQUEST_TIMEOUT, RoutingMode
 from raiden.exceptions import ServiceRequestFailed
+from raiden.network.proxies.service_registry import ServiceRegistry
 from raiden.utils import typing
 from raiden_contracts.utils.proofs import sign_one_to_n_iou
 
+log = structlog.get_logger(__name__)
+
 
 def get_pfs_info(url: str) -> typing.Optional[typing.Dict]:
     try:
-        return requests.get(f'{url}/api/v1/info', timeout=DEFAULT_HTTP_REQUEST_TIMEOUT).json()
+        response = requests.get(
+            f'{url}/api/v1/info',
+            timeout=DEFAULT_HTTP_REQUEST_TIMEOUT,
+        )
+        return response.json()
     except requests.exceptions.RequestException:
         return None
 
 
+def get_random_service(service_registry: ServiceRegistry) -> Optional[str]:
+    latest_block_hash = service_registry.client.blockhash_from_blocknumber(
+        block_number=service_registry.client.block_number(),
+    )
+    count = service_registry.service_count(block_identifier=latest_block_hash)
+    if count == 0:
+        return None
+    index = random.SystemRandom().randint(0, count - 1)
+    address = service_registry.get_service_address(
+        block_identifier=latest_block_hash,
+        index=index,
+    )
+    # We are using the same blockhash for both blockchain queries so the address
+    # should exist for this query. Additionally at the moment there is no way for
+    # services to be removed from the registry.
+    assert address, 'address should exist for this index'
+    return service_registry.get_service_url(
+        block_identifier=latest_block_hash,
+        service_hex_address=address,
+    )
+
+
+def configure_pfs(
+        pfs_address: Optional[str],
+        routing_mode: RoutingMode,
+        service_registry,
+) -> Optional[str]:
+    """
+    Take in the given pfs_address argument, the service registry and find out a
+    pfs address to use.
+
+    If pfs_address is None then basic routing must have been requested.
+    If pfs_address is provided we use that.
+    If pfs_address is 'auto' then we randomly choose a PFS address from the registry
+
+    Returns the PFS url to use or None if we don't use the PFS and use basic routing
+    """
+    if routing_mode == RoutingMode.BASIC:
+        click.secho(
+            'Requested to use basic routing. No path finding service is being used',
```

Comment:
This should also be logged (in order for e.g. debug logs to show the complete state).

And a nitpick: The phrasing sounds slightly awkward. Suggestion: `Not using path finding services, falling back to basic routing.`